src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -1059,6 +1083,104 @@ private void handleThresholdResult() {
         }
     }
 
+    // For single entity detector, receiving results from the ML plugin
+    class PMMLActionListener implements ActionListener<PMMLResultResponse> {
+        private PMMLResultResponse pmmlResult;
+        private List<FeatureData> features;
+        private final String adID;
+        private ActionListener<AnomalyResultResponse> listener;
+        private long detectorIntervalInMinutes;
+
+        PMMLActionListener(
+            List<FeatureData> features,
+            String adID,
+            ActionListener<AnomalyResultResponse> listener,
+            long detectorIntervalInMinutes
+        ) {
+            this.pmmlResult = null;
+            this.features = features;
+            this.adID = adID;
+            this.listener = listener;
+            this.detectorIntervalInMinutes = detectorIntervalInMinutes;
+        }
+
+        @Override
+        public void onResponse(PMMLResultResponse response) {
+            if (response == null) {
+                LOG.warn(NULL_RESPONSE + " ML plugin");
+            } else {
+                pmmlResult = response;
+            }
+            handlePMMLResults();
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            try {
+                handleExecuteException(e, listener, adID);
+            } catch (Exception ex) {
+                LOG.error("Unexpected exception: {} for {}" + ", while calling the ML plugin", ex, adID);
+            } finally {
+                handlePMMLResults();
+            }
+        }
+
+        // Transform result from the ML plugin to the anomaly result format
+        private void handlePMMLResults() {
+            try {
+                if (listener == null) {
+                    LOG.error("Unexpected exception: {} for {}" + ", while calling the ML plugin", "listener is null", adID);
+                    return;
+                }
+                if (pmmlResult == null) {
+                    listener.onFailure(new InternalFailure(adID, "No response from the ML plugin"));
+                    return;
+                }
+                // convert the results from PMML to AD specific format
+                double anomalyGrade = handleGrade(pmmlResult.getOutlier(), pmmlResult.getDecisionFunction());
+                double anomalyScore = handleScore(pmmlResult.getDecisionFunction());
+                // some values are dummy:
+                // 1. Confidence doesn't really change since the ML plugin uploaded models are immutable.
+                double confidence = 1;
+                // 2. PMML models don't have rcfTotalUpdates.
+                long rcfTotalUpdates = 1000;
```

Comment:
Explain more about this hard code value. Current AD needs to calculate RCF model init progress with RCF total updates: 
`init_progress = RCF_total_updates / 128 `

Reply:
Got it

Reply:
Sure, you can add some todo here. No need to fix right now considering your internship will end soon.